### PR TITLE
[netcore] EH work

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by vargaz** [comment]
@monojenkins build failed

File: mcs/class/System.Private.CoreLib/System.Diagnostics/StackTrace.cs [hunk 1]
```diff
@@ -1,20 +1,54 @@
...
+		#region Keep in sync with object-internals.h
+		internal int ilOffset;
+		internal int nativeOffset;
+		internal long methodAddress;
+		internal uint methodIndex;
+		internal MethodBase methodBase;
+		internal string fileName;
+		internal int lineNumber;
+		internal int columnNumber;
+		#pragma warning disable 649
```
Comment (marek-safar) [inline mcs/class/System.Private.CoreLib/System.Diagnostics/StackTrace.cs]: The warning disable can be removed

File: mcs/class/System.Private.CoreLib/System.Diagnostics/StackTrace.cs [hunk 2]
```diff
@@ -1,20 +1,54 @@
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
 namespace System.Diagnostics
 {
+	// Need our own stackframe class since the shared version has its own fields
+	[StructLayout (LayoutKind.Sequential)]
+	class MonoStackFrame {
```
Comment (marek-safar) [inline mcs/class/System.Private.CoreLib/System.Diagnostics/StackTrace.cs]: Would it help if this was struct?

File: mcs/class/System.Private.CoreLib/System.Diagnostics/StackTrace.cs [hunk 3]
```diff
@@ -1,20 +1,54 @@
...
+		#region Keep in sync with object-internals.h
+		internal int ilOffset;
+		internal int nativeOffset;
+		internal long methodAddress;
```
Comment (marek-safar) [inline mcs/class/System.Private.CoreLib/System.Diagnostics/StackTrace.cs]: could you add comment on fields which are unused

File: mcs/class/System.Private.CoreLib/System/Exception.cs [hunk 4]
```diff
@@ -4,8 +4,6 @@ namespace System
 {
 	partial class Exception
 	{
-		protected event EventHandler<SafeSerializationEventArgs> SerializeObjectState;
```
Comment (marek-safar) [inline mcs/class/System.Private.CoreLib/System/Exception.cs]: This API is required

Comment (vargaz) [reply mcs/class/System.Private.CoreLib/System/Exception.cs]: This adds fields to the Exception class whose layout the runtime depends on.

Comment (marek-safar) [reply mcs/class/System.Private.CoreLib/System/Exception.cs]: Then we need to fix it differently, the explicit event probably won't help as the partial order is not guaranteed

File: mcs/class/System.Private.CoreLib/System.Diagnostics/StackTrace.cs (hunk 3 above)
Comment (vargaz) [reply mcs/class/System.Private.CoreLib/System.Diagnostics/StackTrace.cs]: Fixed.

**Comment by vargaz** [comment]
@monojenkins commit apidiff

**Comment by vargaz** [comment]
@monojenkins commit apidiff

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins commit apidiff

**Comment by vargaz** [comment]
@monojenkins commit apidiff

**Comment by akoeplinger** [comment]
@monojenkins commit apidiff

**Comment by akoeplinger** [comment]
The API Diff was already applied with https://github.com/mono/api-snapshot/commit/7896ad7c3d087516f6531926cb1ef1d239f6b1a2

**Pull request merged**